src/dungeon.cpp

```diff
@@ -1299,35 +1299,38 @@ static void commandQuit() {
     }
 }
 
-static void commandPreviousMessage() {
-    int maxMessages = MESSAGE_HISTORY_SIZE;
+static int8_t calculateMaxMessageCount () {
+    int8_t max_messages = MESSAGE_HISTORY_SIZE;
 
     if (command_count > 0) {
-        maxMessages = command_count;
-        if (maxMessages > MESSAGE_HISTORY_SIZE) {
-            maxMessages = MESSAGE_HISTORY_SIZE;
+        max_messages = command_count;
+        if (max_messages > MESSAGE_HISTORY_SIZE) {
+            max_messages = MESSAGE_HISTORY_SIZE;
         }
         command_count = 0;
     } else if (last_command != CTRL_KEY('P')) {
-        maxMessages = 1;
+        max_messages = 1;
     }
+    return max_messages;
+}
 
-    int msgID = last_message_id;
+static void commandPreviousMessage() {
+    const int8_t max_messages = calculateMaxMessageCount();
```

Comment:
The type here doesn't make sense. I also don't see any benefit to using a `const` here.